File: day_1/part1.py
def calculate_fuel(mass):
    return mass // 3 - 2


def _read_masses(path):
    with open(path, "r") as f:
        return list(map(int, f.read().split("\n")))


def caculate_total_fuel(masses):
    return sum(map(calculate_fuel, masses))


if __name__ == "__main__":
    masses = _read_masses("day_1/input.txt")
    fuel_requires = caculate_total_fuel(masses)
    print(len(masses), "masses require", fuel_requires, "fuel")

File: day_1/part1.mjs
import { readFile } from 'fs';
import { join } from 'path';

var filePath = join(process.cwd(), process.argv[2], 'input.txt');

function calculateFuel(mass) {
    return Math.floor(mass / 3) - 2;
}

readFile(filePath, { encoding: 'utf-8' }, (err, data) => {
    if (!err) {
        var fuelRequired = data
            .split("\n")
            .map((fuel) => parseInt(fuel))
            .map(calculateFuel)
            .reduce((total, fuel) => total + fuel);
        console.log("Required fuel:", fuelRequired);
    } else {
        console.log("Error reading file:", err);
    }
});